vm/src/builtins/dict.rs

```diff
@@ -150,14 +150,25 @@ impl PyDict {
         iterable: ArgIterable,
         value: OptionalArg<PyObjectRef>,
         vm: &VirtualMachine,
-    ) -> PyResult<PyRef<Self>> {
+    ) -> PyResult {
+        if !class.is(PyDict::class(vm)) {
+            if let Some(ref slot_new) = class.slots.new.load() {
+                let slot_new_cls = slot_new(class, FuncArgs::default(), vm)?;
```

Comment:
The casue of CI error is lack of calling `__init__`. I found `PyType::call` from `Callable` trait calls both of them.

```suggestion
            let slot_new_cls = PyType::call(&class, ().into(), vm)?;
```

Rather than empty tuple, we can pass `KwArgs` to init dict. 
```suggestion
            let value = value.unwrap_or_none(vm);
            let mut items = IndexMap::<PyObjectRef, PyObjectRef>::new();
            // TODO: fill key-value to items 
            let slot_new_cls = PyType::call(&class, KwArgs::new().into(), vm)?;
```

Reply:
Do you mean kwargs for callable or when invoking fromkeys?

BTW, without load check or some way to distinguish the object, it would recursive call `fromkeys` method till stack overflow… Is there a way to prevent this? Especially when using vm_trace! I run into many times.

Reply:
I think you shouldn't call `fromkeys` inside there. It will be always recursive call in that case.

I checked same funciton in CPython and found my comments about `kwargs` part is wrong. Forget about it.
ref: https://github.com/python/cpython/blob/main/Objects/dictobject.c#L2260-L2351

- It calls the type itself (`_PyObject_CallNoArg(cls)` is `PyType::call(&cls, ().into(), vm)` in RustPython
- If the created object's type is exactly `dict`, it follows the optimized path. (For now, let's regard it same as current implementation)
- Otherwise, it calls setitem over the iterable.

You set items by calling `fromkeys`, but it seems calling setitem for every item is the way how it does.

Reply:
It make sense. Thanks for your advise!